lib/executor/variable_arrival_rate.go

```diff
@@ -390,10 +390,19 @@ func (varr VariableArrivalRate) Run(ctx context.Context, out chan<- stats.Sample
 	go trackProgress(ctx, maxDurationCtx, regDurationCtx, varr, progresFn)
 
 	regDurationDone := regDurationCtx.Done()
-	runIterationBasic := getIterationRunner(varr.executionState, varr.logger, out)
-	runIteration := func(vu lib.VU) {
+	runIterationBasic := getIterationRunner(varr.executionState, varr.logger)
+	runIteration := func(initVU lib.InitializedVU, modActive bool) {
+		ctx, cancel := context.WithCancel(maxDurationCtx)
+		defer cancel()
+
+		vu := initVU.Activate(&lib.VUActivationParams{RunContext: ctx})
```

Comment:
Again, activation should happen only once here, when you get the VU from the pile of initialized VUs.

Reply:
Should be resolved by 9031ad21.